include/emds/emds.h

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file
+ *  @defgroup emds Emergency Data Storage
+ *  @ingroup emds
+ *  @{
+ *  @brief Emergency Data Storage API
+ */
+
+#ifndef EMDS_H__
+#define EMDS_H__
+
+#include <stddef.h>
+#include <sys/types.h>
+#include <sys/util.h>
+#include <sys/slist.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @struct emds_entry
+ *
+ * Information about entries used to store data in the emergency data storage.
+ */
+struct emds_entry {
+	/** Unique id for each static and dynamic entry. */
+	uint16_t id;
+	/** Pointer to data that shall be stored. */
+	uint8_t *data;
+	/** Length of data that shall be stored. */
+	size_t len;
+}
+
+/**
+ * @struct emds_dynamic_entry
+ *
+ * Entries with a node element, used for dynamic entries. These are registered
+ * using a call to emds_entry_add().
+ */
+struct emds_dynamic_entry {
+	struct emds_entry entry;
+	sys_snode_t node;
+}
+
+/**
+ * @brief Define a static entry for emergency data storage items.
+ *
+ * @param name The entry name.
+ * @param _id Unique id for entry. This value and not overlap with any other
+ *            value.
+ * @param _data Data pointer to be stored at emergency data store.
+ * @param _len Length of data to be stored at emergency data store.
+ *
+ * This creates a variable _name prepended by emds_.
+ */
+#define EMDS_STATIC_ENTRY_DEFINE(_name, _id, _data, _len)                      \
+	(const STRUCT_SECTION_ITERABLE(emds_entry, emds_##_name) = {           \
+		.id = _id,                                                     \
+		.data = (uint8_t *)_data,                                      \
+		.len = _len,                                                   \
+	})
+
+/**
+ * @brief Initialize the emergency data storage.
+ *
+ * Initializes the emergency data storage. This need to called before starting
+ * to add entries to the emds_dynamic_entry and doing restore.
+ *
+ * @retval 0 Success
+ * @retval -ERRNO errno code if error
+ */
+int emds_init(void);
+
+/**
+ *  @brief Add entry that should be saved/restored when emergency data storage is called.
+ *
+ * This will add the entry to the dynamic entry list, and will when the
+ * 'emds_store' function is called take the given data pointer and store the
```

Comment:
```suggestion
 * @ref emds_store function is called take the given data pointer and store the
```
Not sure if it is still valid syntax.